matrix_client/api.py

```diff
@@ -774,8 +776,74 @@ def get_download_url(self, mxcurl):
         else:
             raise ValueError("MXC URL did not begin with 'mxc://'")
 
+    def media_download(self, mxcurl):
+        """Download raw media from provided mxc URL.
+
+        Args:
+            mxcurl (str): mxc media URL
+        """
+        if mxcurl.startswith('mxc://'):
+            return self._send(
+                "GET", mxcurl[6:],
+                api_path="/_matrix/media/r0/download/",
+                return_json=False
+            )
+        else:
+            raise ValueError(
+                "MXC URL '%s' did not begin with 'mxc://'" % mxcurl
+            )
+
+    def get_thumbnail(self, mxcurl, width, height, method='scale'):
+        """Download raw media thumbnail from provided mxc URL.
+
+        Args:
+            mxcurl (str): mxc media URL
+            width (int): required thumbnail width
+            height (int): required thumbnail height
```

Comment:
Might as well say "desired" instead of "required" here since the spec insists on this.